### PR TITLE
resolve python error 'tuple parameter unpacking is not supported in p…

### PR DESCRIPTION
**Pull request by hougomartim**
…ython3' in '_find_delay_in_mutant' function ./

### CODE REVIEW

**Comment by andresriancho** [comment]
@hougomartim are you trying to migrate w3af to py3k? I'm curious, how is that going? What works? What does not work?

File: w3af/core/controllers/core_helpers/consumers/tests/test_base_consumer.py [hunk 1]
```diff
@@ -39,15 +39,15 @@ def test_handle_exception(self):
...
+        except Exception as e:
             self.bc.handle_exception('audit', 'sqli', fr, e)
 
         exception_data = self.bc.out_queue.get()
 
         self.assertTrue(exception_data.traceback is not None)
         self.assertEqual(exception_data.phase, 'audit')
         self.assertEqual(exception_data.plugin, 'sqli')
-        self.assertEqual(exception_data.exception, e)
+        self.assertEqual(exception_data.Exception as e)
```
Comment (andresriancho) [inline w3af/core/controllers/core_helpers/consumers/tests/test_base_consumer.py]: Nope, this is a bug.

File: w3af/core/controllers/core_helpers/exception_handler.py [hunk 2]
```diff
@@ -73,7 +73,7 @@ def handle_exception_data(self, exception_data):
                     (_, _, exception_data.traceback),
                     exception_data.enabled_plugins)
 
-    def handle(self, current_status, exception, exec_info, enabled_plugins):
+    def handle(self, current_status, Exception as exec_info, enabled_plugins):
```
Comment (andresriancho) [inline w3af/core/controllers/core_helpers/exception_handler.py]: Bug

**Comment by andresriancho** [comment]
Spotted a few bugs during the review, maybe you should try running the command referenced in here?

https://portingguide.readthedocs.io/en/latest/exceptions.html#the-new-except-syntax

That might produce a better result with the except syntax porting.

File: w3af/core/ui/api/tests/test_exceptions.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def test_query_exceptions(self):
                             #u'lineno': 123,
                             u'phase': u'phase',
                             u'plugin': u'plugin'}
-        self.assertEqual(exception, expected_summary)
+        self.assertEqual(Exception as expected_summary)
```
Comment (andresriancho) [inline w3af/core/ui/api/tests/test_exceptions.py]: Bug

**Comment by andresriancho** [comment]
w3af embeds a `sqlmap` version in `w3af/plugins/attack/db/sqlmap/`. Every couple of months I copy the latest sqlmap from their repository into w3af's repo.

If we apply the changes you propose to that directory then the merge / copy I perform every couple of months will fail (merge conflict) or overwrite the changes.

Not sure how to solve this problem.

File: w3af/plugins/audit/redos.py [hunk 4]
```diff
@@ -59,7 +59,7 @@ def _generate_delay_tests(self, freq, debugging_id):
             for delay_obj in self.get_delays():
                 yield mutant, delay_obj, debugging_id
 
-    def _find_delay_in_mutant(self, (mutant, delay_obj, debugging_id)):
+    def _find_delay_in_mutant(self, mutant, delay_obj, debugging_id):
```
Comment (andresriancho) [inline w3af/plugins/audit/redos.py]: And are you sure that the plugin still works as expected? Those () were there for a reason :-)

File: w3af/plugins/tests/crawl/phpinfo/phpinfo-5.1.3-rc4dev.html [hunk 5]
```diff
@@ -391,7 +391,7 @@ <h2><a name="module_SPL">SPL</a></h2>
...
+<tr><td class="e">Classes </td><td class="v">AppendIterator, ArrayIterator, ArrayObject, BadFunctionCallException, BadMethodCallException, CachingIterator, DirectoryIterator, DomainException as emptyIterator, FilterIterator, InfiniteIterator, InvalidArgumentException, IteratorIterator, LengthException, LimitIterator, LogicException, NoRewindIterator, OutOfBoundsException, OutOfRangeException, OverflowException, ParentIterator, RangeException, RecursiveArrayIterator, RecursiveCachingIterator, RecursiveDirectoryIterator, RecursiveFilterIterator, RecursiveIteratorIterator, RuntimeException, SimpleXMLIterator, SplFileInfo, SplFileObject, SplObjectStorage, SplTempFileObject, UnderflowException, UnexpectedValueException </td></tr>
```
Comment (andresriancho) [inline w3af/plugins/tests/crawl/phpinfo/phpinfo-5.1.3-rc4dev.html]: Bug

File: w3af/plugins/tests/crawl/phpinfo/phpinfo-5.1.6.html [hunk 6]
```diff
@@ -470,7 +470,7 @@ <h2><a name="module_SPL">SPL</a></h2>
...
+<tr><td class="e">Classes </td><td class="v">AppendIterator, ArrayIterator, ArrayObject, BadFunctionCallException, BadMethodCallException, CachingIterator, DirectoryIterator, DomainException as emptyIterator, FilterIterator, InfiniteIterator, InvalidArgumentException, IteratorIterator, LengthException, LimitIterator, LogicException, NoRewindIterator, OutOfBoundsException, OutOfRangeException, OverflowException, ParentIterator, RangeException, RecursiveArrayIterator, RecursiveCachingIterator, RecursiveDirectoryIterator, RecursiveFilterIterator, RecursiveIteratorIterator, RuntimeException, SimpleXMLIterator, SplFileInfo, SplFileObject, SplObjectStorage, SplTempFileObject, UnderflowException, UnexpectedValueException </td></tr>
```
Comment (andresriancho) [inline w3af/plugins/tests/crawl/phpinfo/phpinfo-5.1.6.html]: Bug

**Review by andresriancho — CHANGES_REQUESTED** [review]
A few bugs here and there

File: w3af/plugins/audit/redos.py (hunk 4 above)
Comment (hougomartim) [reply w3af/plugins/audit/redos.py]: i think that not working. are you sure?

**Pull request closed without merging**